File: Java/LinearSearch.java
package com.code2hell;


import java.util.Scanner;



public class LinearSearch{

//    method
    public static int linearSearch(int[] arr, int key){
        for(int i=0;i<arr.length;i++){
            if(arr[i] == key){
                return i;
            }
        }
        return -1;
    }

//    code Derived
    public static void main(String a[]){
        int[] a1= {10,20,30,50,70,90};
        int key = 50;
        System.out.println(key+" is found at index: "+linearSearch(a1, key));
    }
}



//using scanner function

class LinearSearchExample2
{
    public static void main(String args[])
    {
        int c, n, search, array[];

        Scanner in = new Scanner(System.in);
        System.out.println("Enter number of elements");
        n = in.nextInt();
        array = new int[n];

        System.out.println("Enter those " + n + " elements");

        for (c = 0; c < n; c++)
            array[c] = in.nextInt();

        System.out.println("Enter value to find");
        search = in.nextInt();

        for (c = 0; c < n; c++)
        {
            if (array[c] == search)     /* Searching element is present */
            {
                System.out.println(search + " is present at location " + (c + 1) + ".");
                break;
            }
        }
        if (c == n)  /* Element to search isn't present */
            System.out.println(search + " isn't present in array.");
    }
}











// Java code for linearly searching x in arr[]. If x
// is present then return its location, otherwise
// return -1

class GFG
{
    public static int search(int arr[], int x)
    {
        int n = arr.length;
        for (int i = 0; i < n; i++)
        {
            if (arr[i] == x)
                return i;
        }
        return -1;
    }

    // Driver code
    public static void main(String args[])
    {
        int arr[] = { 2, 3, 4, 10, 40 };
        int x = 10;

        // Function call
        int result = search(arr, x);
        if (result == -1)
            System.out.print(
                    "Element is not present in array");
        else
            System.out.print("Element is present at index "
                    + result);
    }
}

File: Python/LinearSearch.py
# def LinearSearch(list1, element):
#     for i in range(len(list1)):   #i used to go each index of a given list
#         if(element==list1[i]):
#             print("Element is found at index", i)
#             break
#     else:
#         print("Element is not found !")

# list1= [3,4,5,7,8,87,35,78,]
# element= 23

# print(LinearSearch(list1, element))     



# search repeated element in list 

def RLSearch(list1, element2):
    list2 = []
    flag = False

    for i in range(len(list1)):
        if(element2==list1[i]):
            flag = True
            list2.append(i)

    if(flag == True):
        for i in list2:
            print("Element",element2,"is found at index: ",i)

    else:
        print("Element is not found !")

list2= [3,4,5,7,3,3,8,87,35,3,78,]
element2= int(input("Enter the number: "))

print(RLSearch(list2, element2)) 

